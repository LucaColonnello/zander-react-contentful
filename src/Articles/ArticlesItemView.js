import React from 'react';
// import dashify from 'dashify'
import { Link } from 'react-router-dom';
import months from '../utils/months';

const ArticlesItemView = props => {
  const { title, subtitle, date, id } = props;
  const formattedDate = new Date(date);
  // const url = `/articles/${dashify(title)}`
  const url = `/articles/${id}`;
  return (
    <li className="contentList-group-items-item">
      <Link className="infoLink" to={url} title={'Read ' + title}>
        <div className="infoLink-content">
          <div className="infoLink-content-title">{title}</div>
          {subtitle
            ? <div className="infoLink-content-excerpt">{subtitle}</div>
            : null}
        </div>
        <time className="infoLink-date">
          {formattedDate.getDay()}
          {' '}
          {months[formattedDate.getMonth()]}
          {' '}
          {formattedDate.getUTCFullYear()}
        </time>
      </Link>
    </li>
  );
};

export default ArticlesItemView;
