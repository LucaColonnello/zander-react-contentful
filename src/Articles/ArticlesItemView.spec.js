import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ArticlesItemView from './ArticlesItemView';

test('Article view snapshot test', () => {
  const component = shallow(
    <ArticlesItemView
      title="The title"
      subtitle="The subtitle"
      body="The body"
      date="2017-02-02T00:00+00:00"
    />,
  );
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});
