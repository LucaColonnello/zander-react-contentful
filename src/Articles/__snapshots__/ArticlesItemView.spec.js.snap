// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article view snapshot test 1`] = `
<li
  className="contentList-group-items-item"
>
  <Link
    className="infoLink"
    replace={false}
    title="Read The title"
    to="/articles/undefined"
  >
    <div
      className="infoLink-content"
    >
      <div
        className="infoLink-content-title"
      >
        The title
      </div>
      <div
        className="infoLink-content-excerpt"
      >
        The subtitle
      </div>
    </div>
    <time
      className="infoLink-date"
    >
      4
       
      Feb
       
      2017
    </time>
  </Link>
</li>
`;
